legate/legate_kvikio/tile.py

```diff
@@ -102,3 +102,66 @@ def read_tiles(
         The start coordinate of the tiles
     """
     return _tile_read_write(TaskOpCode.TILE_READ, ary, dirpath, tile_shape, tile_start)
+
+
+def read_tiles_by_offsets(
+    ary: cunumeric.ndarray,
+    filepath: Iterable[pathlib.Path | str],
+    offsets: Tuple[int],
+    sizes: Tuple[int],
+    tile_shape: Tuple[int],
+) -> None:
+    """Read multiple tiles from a single file into an array using KvikIO
+
+    The array shape must be divisible with the tile shape.
+
+    # TODO: support a filepath per offset/size
+
+    Parameters
+    ----------
+    ary
+       The cuNumeric array to read into.
+    filepath
+        Filepath to the file.
+    offsets
+        The offset of each tile in the file (in bytes).
+    sizes
+        The size of each tile in the file (in bytes).
+    tile_shape
+        The shape of each tile.
+    """
+
+    if any(d % c != 0 for d, c in zip(ary.shape, tile_shape)):
```

Comment:
Does the number of modes of the `tile_shape` have to match the number of modes of the array? If so maybe add

```
if len(ary.shape) != len(tile_shape):
    raise ValueError("Tile shape and array shape must have  same number of axes")
```

Reply:
Done